File: Algorithms/27. Remove Element/golang/solution.go
func removeElement(nums []int, val int) int {
    n := len(nums)
	last := 0

    for i := 0; i < n; i++ {
        if nums[i] != val {
            nums[last] = nums[i]
            last++
        }
    }

    return last
}

File: Algorithms/27. Remove Element/javascript/solution.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
    let i = 0;
    for (const j of nums) {
        if (j !== val) {
            nums[i] = j;
            i++;
        }
    }
    return i;
};
